subsys/bootloader/provision/provision_flash.c

```diff
@@ -37,34 +51,72 @@ u32_t num_public_keys_read(void)
 	return p_provision_data->num_public_keys;
 }
 
+static const u32_t *get_key(u32_t key_idx)
+{
+	return &(p_provision_data->key_data[key_idx *
+			(CONFIG_SB_PUBLIC_KEY_HASH_LEN / 4 + 1)]);
+}
+
+#define INVALID_VAL 0xFFFF0000
+
+static const u32_t *is_valid(u32_t key_idx)
+{
+	return get_key(key_idx);
+}
+
 int public_key_data_read(u32_t key_idx, u32_t *p_buf, size_t buf_size)
 {
 	const u32_t *p_key;
 
+	if (*is_valid(key_idx) == INVALID_VAL) {
+		return -EINVAL;
+	}
+
 	if (buf_size < CONFIG_SB_PUBLIC_KEY_HASH_LEN) {
 		return -ENOMEM;
 	}
 
-	if (key_idx >= p_provision_data->num_public_keys) {
-		return -EINVAL;
+	if (key_idx >= num_public_keys_read()) {
+		return -EFAULT;
 	}
 
-	p_key = &(p_provision_data->pkd[key_idx *
-			CONFIG_SB_PUBLIC_KEY_HASH_LEN / 4]);
+	p_key = get_key(key_idx) + 1;
 
-#ifdef CONFIG_SOC_NRF9160
 	/* Ensure word alignment, as provision data is stored in memory region
 	 * with word sized read limitation. Perform both build time and run
 	 * time asserts to catch the issue as soon as possible.
 	 */
 	BUILD_ASSERT(CONFIG_SB_PUBLIC_KEY_HASH_LEN % 4 == 0);
-	BUILD_ASSERT(offsetof(provision_flash_t, pkd) % 4 == 0);
+	BUILD_ASSERT(offsetof(provision_flash_t, key_data) % 4 == 0);
 	__ASSERT(((u32_t)p_key % 4 == 0), "Key address is not word aligned");
-#endif /* CONFIG_SOC_NRF9160 */
-
 	for (size_t i = 0; i < CONFIG_SB_PUBLIC_KEY_HASH_LEN/4; i++) {
 		p_buf[i] = p_key[i];
 	}
 
 	return CONFIG_SB_PUBLIC_KEY_HASH_LEN;
 }
+
+void invalidate_public_key(u32_t key_idx)
```

Comment:
Will this be the same for the keys stored in the OTP region?

Reply:
Yes